Dockerfile

```diff
@@ -30,78 +33,93 @@ LABEL org.label-schema.vcs-url="https://github.com/usdot-fhwa-stol/carma-platfor
-        DEBIAN_FRONTEND=noninteractive apt-get install -y \
-        git \
-        ssh \
-        ros-kinetic-rosbridge-server \
-        sudo \
-        tmux \
-        vim \
-        nano \
-        less \
-        curl \
+# Install ROS Noetic
+ARG ROS_DISTRO=noetic
+RUN sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' \ 
+    && apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 \ 
+    && apt-get update \ 
+    && apt-get install ros-noetic-desktop-full python3-rosinstall -y
+
+# Prepare for ROS 2 Foxy installation
+RUN apt update && apt install locales \
+    && locale-gen en_US en_US.UTF-8 \
+    && update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 \
+    && export LANG=en_US.UTF-8
+RUN apt update && apt install curl gnupg2 lsb-release \
+    && curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | apt-key add -
+RUN sh -c 'echo "deb [arch=$(dpkg --print-architecture)] http://packages.ros.org/ros2/ubuntu $(lsb_release -cs) main" > /etc/apt/sources.list.d/ros2-latest.list'
+
+# Install ROS 2 Foxy
+RUN apt update && apt install ros-foxy-desktop -y
+
+RUN apt-get update && apt-get install -y \
         apt-transport-https \
-        python-catkin-pkg \
-        python-rosdep \
-        python-pip \
-        python3-pip \
-        python3-colcon-common-extensions \
-        python3-setuptools \
-        python3-vcstool \
-        nodejs \
-        unzip \
+        apt-utils \
+        automake \
+        autotools-dev \
+        curl \
+        dialog \
         gcovr \
-        libpcap-dev \
-        libfftw3-dev \
+        gdb \
+        git \
         gnuplot-qt \
+        less \
+        libboost-python-dev \
+        libfftw3-dev \
         libgeographic-dev \ 
+        libgmock-dev \
+        libpcap-dev \
         libpugixml-dev \
-        python-catkin-tools \
-        libboost-python-dev \
-        sqlite3 \
-        autotools-dev \
-        automake \
-        ros-kinetic-rosserial-arduino \
-        ros-kinetic-rosserial \
-        dialog \
-        x-window-system \
         mesa-utils \
-        gdb \
-        software-properties-common
-
+        nano \
+        nodejs \
+        python3-catkin-pkg \
+        python3-catkin-tools \
+        python3-colcon-common-extensions \
+        python3-pip \
+        python3-rosdep \
+        python3-setuptools \
+        python3-vcstool \
+        ros-noetic-costmap-2d \
+        ros-noetic-geodesy \
+        ros-noetic-rosbridge-server \
+        ros-noetic-rosserial \
+        ros-noetic-rosserial-arduino \
+        software-properties-common \
+        sqlite3 \
+        ssh \
+        sudo \
+        tmux \
+        unzip \
+        vim \
+        x-window-system
 
-RUN pip3 install -U setuptools
+RUN pip3 install -U testresources setuptools
 
 # Install simple-pid
-RUN pip install simple-pid
+RUN pip3 install simple-pid
```

Comment:
In ubuntu 20 shouldn't the default python be 3? Is it necessary to change pip to pip3?

Reply:
Yes, fixed.